docs/DeveloperGuide.md

```diff
@@ -1,37 +1,180 @@
+`AnswerTracker`. Both are used to display the results for
+all question sets attempted by the user, including details
+such as specific questions and their respective answer inputs
+from the user.
+
+Given below is an example usage scenario and how the results
+mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time,
+and proceeds to start a game with their chosen topic.
+
+The following sequence diagram shows how the `Results` for
+one question set is added to the `ResultsList`:  
+![ResultsList sequence diagram](./team/img/Results.png)
+> **Note:** The lifeline for Parser and Results should end
+> at the destroy marker (X) but due to a limitation of PlantUML,
+> the lifeline reaches the end of the diagram.
+
+Similarly, the following sequence diagram shows how the
+`AnswerTracker` stores all the user answer inputs:  
+![AnswerTracker sequence diagram](./team/img/AnswerTracker.png)
+> **Note:** The lifeline for Parser, allAnswers, and answersCorrectness
+> should end
+> at the destroy marker (X) but due to a limitation of PlantUML,
+> the lifeline reaches the end of the diagram.
+
+Step 2. The user may repeat Step 1 with other question sets.  
+
+The sequence diagram for this step involving `ResultsList` 
+and `AnswerTracker` are the same as the ones shown in Step 1.
+
+Step 3. The user now wants to view their results by executing
+the `results` command.  
+![Results command sequence diagram](./team/img/ViewResults.png)
+
+> **Note:** If the user uses the results feature before
+> attempting any questions, the application will instead 
+> return an error to the user indicating that there are no 
+> results.
+
+
+### Topics Feature
+
+The topics feature comprises `TopicList` and `QuestionListByTopic`. 
+`TopicList` is the list of topics for the users to attempt.
+`QuestionListByTopic` stores the respective question set for each topic in an ArrayList. 
+
+Given below is an example usage scenario and how the results
+mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time,
+and proceeds to start a game with their chosen topic.
+
+The following shows the class diagram for `topicList`:
+![TopicList class diagram](./team/img/TopicList_Topic_class_diagram.png)
+
+Step 2. A question from the question set of the chosen topic is displayed.
+        The user inputs their answer.
+
+Step 3. Step 2 repeats until all the questions in the question set has been asked.
+        Step 1 executes and process repeats.
+
+The following shows the class diagram for `QuestionListByTopic`:
+![QuestionListByTopic](./team/img/QuestionListByTopic_class_diagram.png)
+
+### Solution feature
+
+The solution feature either prints the solution to 1 question or all questions in 1 topic.
+
+The solution feature is facilitated by `Parser#processSolutionCommand`, which is called by `Parser#parseCommand`
+
+> **OVERVIEW:**
+> ![Solution sequence diagram](./team/img/Solution.png)
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111226733/a02973c2-fed4-4c95-8b98-35001ff71ffa)


Have you maybe forgot to add [else] int the alt block?